src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -152,28 +160,49 @@ private void filterCaseDataBySecurityClassification(CaseDetails caseDetails) {
     private void filterCaseDataForMultiCaseTypeSearch(CrossCaseTypeSearchRequest searchRequest, CaseType authorisedCaseType, CaseDetails caseDetails) {
         if (searchRequest.isMultiCaseTypeSearch() && caseDetails.getData() != null) {
             JsonNode caseData = caseDataToJsonNode(caseDetails);
+            String caseDataJson = caseData.toString();
             JsonNode filteredMultiCaseTypeSearchData = objectMapperService.createEmptyJsonNode();
 
             authorisedCaseType.getSearchAliasFields()
                 .stream()
                 .filter(searchRequest::hasAliasField)
-                .forEach(searchAliasField -> findCaseFieldPathInCaseData(caseData, searchAliasField.getCaseFieldPath())
+                .forEach(searchAliasField -> findCaseFieldPathInCaseData(authorisedCaseType, caseDataJson, searchAliasField.getCaseFieldPath())
                     .filter(not(JsonNode::isMissingNode))
                     .ifPresent(jsonNode -> ((ObjectNode) filteredMultiCaseTypeSearchData).set(searchAliasField.getId(), jsonNode)));
 
             caseDetails.setData(jsonNodeToCaseData(filteredMultiCaseTypeSearchData));
         }
     }
 
-    private Optional<JsonNode> findCaseFieldPathInCaseData(JsonNode caseData, String path) {
-        String jsonPointerExpr;
-        if (path.contains(DOT_SEPARATOR)) {
-            jsonPointerExpr = JSON_EXPR_LOGICAL_SEPARATOR + path.replaceAll(DOT_SEPARATOR_REGEX, JSON_EXPR_LOGICAL_SEPARATOR);
-        } else {
-            jsonPointerExpr = JSON_EXPR_LOGICAL_SEPARATOR + path;
+    private Optional<JsonNode> findCaseFieldPathInCaseData(CaseType caseType, String caseDataJson, String path) {
+        try {
+            String fieldPath = JSON_PATH_ROOT_ELEMENT_PREFIX + sanitiseCollectionFieldInPath(caseType, path);
+            return of(ofNullable(JsonPath.parse(caseDataJson).read(fieldPath, JsonNode.class))
```

Comment:
Using jsonpath parser in place of JsonNode at method. This has rich support and fixes issue with collection field lookups